server/routes/dsl_route.ts

```diff
@@ -141,19 +154,32 @@ export function registerDslRoute(router: IRouter) {
   // Get Indices
   router.get(
     {
-      path: ServiceEndpoints.GetIndexes,
-      validate: {},
+      path: `${ServiceEndpoints.GetIndexes}/{dataSourceId?}`,
+      validate: {
+        params: schema.object({
+          dataSourceId: schema.maybe(schema.string({ defaultValue: '' }))
+        }),
+      },
     },
     async (context, request, response) => {
       const params = {
         format: 'json',
       };
       const start = performance.now();
       try {
-        const resp = await context.core.opensearch.legacy.client.callAsCurrentUser(
-          'cat.indices',
-          params
-        );
+        let resp;
+        const dataSourceId  = request.params.dataSourceId;
+          if(dataSourceEnabled && dataSourceId){
+            let client = await context.dataSource.opensearch.legacy.getClient(dataSourceId);
+            
+            resp = await client.callAPI('cat.indices', params);
+          }
+          else{
+              resp = await context.core.opensearch.legacy.client.callAsCurrentUser(
+                'cat.indices',
+                params
+              );
+          }
```

Comment:
not an issue but would this be better? so that args of api call are not duplicated
```suggestion
        const dataSourceId  = request.params.dataSourceId;
          let callApi: ILegacyScopedClusterClient['callAsCurrentUser'];
          if(dataSourceEnabled && dataSourceId){
            callApi = context.dataSource.opensearch.legacy.getClient(dataSourceId1).callAPI;
          } else {
            callApi = context.core.opensearch.legacy.client.callAsCurrentUser;
          }
          const resp = await callApi('cat.indices', params);
```